common/environment/environment.cpp

```diff
@@ -1336,12 +1536,221 @@ void CLocalEnvironment::clearCache()
         p.setown(conn->getRoot());
     }
     cache.kill();
-    machineCacheBuilt = false;
-    dropZoneCacheBuilt = false;
+    numOfDropzonesByComputer.clear();
+    init();
     resetPasswordsFromSDS();
 }
 
+IConstDropZoneInfo * CLocalEnvironment::findDropZoneByAddress(const char * netaddress, const char *targetFilePath) const
```

Comment:
minor: all other methods returned a linked item. So perhaps this should be : getDropZoneByAddressPath(